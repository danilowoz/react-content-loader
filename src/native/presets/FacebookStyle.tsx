import * as React from 'react'

import { IContentLoaderProps } from '..'
import ContentLoader, { Circle, Rect } from '../ContentLoader'

const ReactContentLoaderFacebook: React.FC<IContentLoaderProps> = props => (
  <ContentLoader viewBox="0 0 476 124" width={476} height={124} {...props}>
    <Rect x="48" y="8" width="88" height="6" rx="3" />
    <Rect x="48" y="26" width="52" height="6" rx="3" />
    <Rect x="0" y="56" width="410" height="6" rx="3" />
    <Rect x="0" y="72" width="380" height="6" rx="3" />
    <Rect x="0" y="88" width="178" height="6" rx="3" />
    <Circle cx="20" cy="20" r="20" />
  </ContentLoader>
)

export default ReactContentLoaderFacebook
