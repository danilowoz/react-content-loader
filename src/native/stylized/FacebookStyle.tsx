import * as React from 'react'

import Holder, { Rect, Circle } from '../Holder'
import { IContentLoaderProps } from '../'

export default (props: IContentLoaderProps) => (
  <Holder {...props}>
    <Rect x="70" y="15" rx="4" ry="4" width="117" height="6" />
    <Rect x="70" y="35" rx="3" ry="3" width="85" height="6" />
    <Rect x="0" y="80" rx="3" ry="3" width="350" height="6" />
    <Rect x="0" y="100" rx="3" ry="3" width="380" height="6" />
    <Rect x="0" y="120" rx="3" ry="3" width="201" height="6" />
    <Circle cx="30" cy="30" r="30" />
  </Holder>
)
