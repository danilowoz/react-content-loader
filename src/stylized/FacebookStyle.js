//@flow
import * as React from 'react'
import Wrap from '../Wrap'
import type { WrapProps } from '../Wrap'

const FacebookStyle = (props: WrapProps): React.Element<*> => (
  <Wrap {...props}>
	<rect x="70" y="15.82" rx="4" ry="4" width="117" height="6.4" />
	<rect x="70" y="36.05" rx="3" ry="3" width="85" height="6.4" />
	<rect x="2" y="81" rx="3" ry="3" width="350" height="6.4" />
	<rect x="0" y="102.38" rx="3" ry="3" width="380" height="6.4" />
	<rect x="0" y="123.26" rx="3" ry="3" width="201.6" height="6.4" />
	<circle cx="30.284271247461902" cy="29.334271247461903" r="28.284271247461902" />
  </Wrap>
)

export default FacebookStyle
