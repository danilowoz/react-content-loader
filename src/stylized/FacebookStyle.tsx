import * as React from 'react'

import Holder from '../Holder'
import { IContentLoaderProps } from '../'

const ReactContentLoaderFacebook: React.FC<IContentLoaderProps> = props => (
  <Holder {...props}>
    <rect x="70" y="15" rx="4" ry="4" width="117" height="6.4" />
    <rect x="70" y="35" rx="3" ry="3" width="85" height="6.4" />
    <rect x="0" y="80" rx="3" ry="3" width="350" height="6.4" />
    <rect x="0" y="100" rx="3" ry="3" width="380" height="6.4" />
    <rect x="0" y="120" rx="3" ry="3" width="201" height="6.4" />
    <circle cx="30" cy="30" r="30" />
  </Holder>
)

export default ReactContentLoaderFacebook
