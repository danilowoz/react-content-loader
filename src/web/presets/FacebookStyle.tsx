import * as React from 'react'

import { IContentLoaderProps } from '..'
import ContentLoader from '../ContentLoader'

const ReactContentLoaderFacebook: React.FC<IContentLoaderProps> = props => (
  <ContentLoader viewBox="0 0 476 124" {...props}>
    <rect x="48" y="8" width="88" height="6" rx="3" />
    <rect x="48" y="26" width="52" height="6" rx="3" />
    <rect x="0" y="56" width="410" height="6" rx="3" />
    <rect x="0" y="72" width="380" height="6" rx="3" />
    <rect x="0" y="88" width="178" height="6" rx="3" />
    <circle cx="20" cy="20" r="20" />
  </ContentLoader>
)

export default ReactContentLoaderFacebook
