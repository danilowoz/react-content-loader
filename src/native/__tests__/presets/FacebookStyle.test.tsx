import 'react-native'

import * as React from 'react'
import * as renderer from 'react-test-renderer'

import FacebookStyle from '../../presets/FacebookStyle'

describe('FacebookStyle', () => {
  const wrapper = renderer.create(
    <FacebookStyle uniqueKey="FacebookStyle" animate={false} speed={20} />
  )

  test('renders correctly', () => {
    const tree = wrapper.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('props are propagated ', () => {
    expect(wrapper.root.props.speed).toEqual(20)
  })
})
