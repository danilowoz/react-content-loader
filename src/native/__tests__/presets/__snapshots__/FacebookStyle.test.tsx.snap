// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacebookStyle renders correctly 1`] = `
<Svg
  height={124}
  style={Object {}}
  viewBox="0 0 476 124"
  width={476}
>
  <Rect
    clipPath="url(#FacebookStyle-animated-diff)"
    fill="url(#FacebookStyle-diff)"
    height="100%"
    width="100%"
    x="0"
    y="0"
  />
  <Defs>
    <ClipPath
      id="FacebookStyle-animated-diff"
    >
      <Rect
        height="6"
        rx="3"
        width="88"
        x="48"
        y="8"
      />
      <Rect
        height="6"
        rx="3"
        width="52"
        x="48"
        y="26"
      />
      <Rect
        height="6"
        rx="3"
        width="410"
        x="0"
        y="56"
      />
      <Rect
        height="6"
        rx="3"
        width="380"
        x="0"
        y="72"
      />
      <Rect
        height="6"
        rx="3"
        width="178"
        x="0"
        y="88"
      />
      <Circle
        cx="20"
        cy="20"
        r="20"
      />
    </ClipPath>
    <LinearGradient
      id="FacebookStyle-diff"
      x1="-100%"
      x2="100%"
      y1={0}
      y2={0}
    >
      <Stop
        offset={0}
        stopColor="#f5f6f7"
      />
      <Stop
        offset={0}
        stopColor="#eee"
      />
      <Stop
        offset={0}
        stopColor="#f5f6f7"
      />
    </LinearGradient>
  </Defs>
</Svg>
`;
